src/node.cc

```diff
@@ -1436,7 +1436,17 @@ void FatalException(Isolate* isolate,
       exit(7);
     } else if (caught->IsFalse()) {
       ReportException(env, error, message);
-      exit(1);
+
+      // fatal_exception_function call before may have set a new exit code ->
+      // read it again, otherwise use default for uncaughtException 1
+      Local<String> exit_code = env->exit_code_string();
+      int code = process_object->Get(env->context(), exit_code)
+        .ToLocalChecked()->Int32Value(env->context()).ToChecked();
+      if (code == 0) {
```

Comment:
Hmm... this should likely only emit `exit(1)` if `code` is `undefined`. It should be legitimate for an uncaught exception handler to set `process.exitCode = 0` and have that be the actual exit code.